tcp_check/datadog_checks/tcp_check/tcp_check.py

```diff
@@ -85,25 +189,51 @@ def _check(self, instance):
                                  'than the system tcp stack allows. You might want to '
                                  'change this setting to allow longer timeouts')
                 self.log.info("System tcp timeout. Assuming that the checked system is down")
-                return Status.DOWN, """Socket error: {}.
+                status_checks.append((
+                    self.SERVICE_CHECK_CAN_CONNECT,
+                    Status.DOWN,
+                    """Socket error: {}.
                  The connection timed out after {} ms because it took more time than the system tcp stack allows.
-                 You might want to change this setting to allow longer timeouts""".format(str(e), length)
+                 You might want to change this setting to allow longer timeouts""".format(str(e), length)))
             else:
                 self.log.info("{}:{} is DOWN ({}). Connection failed after {} ms".format(addr, port, str(e), length))
-                return Status.DOWN, "{}. Connection failed after {} ms".format(str(e), length)
+                status_checks.append((
+                    self.SERVICE_CHECK_CAN_CONNECT,
+                    Status.DOWN,
+                    "{}. Connection failed after {} ms".format(str(e), length)))
 
         except Exception as e:
             length = int((time.time() - start) * 1000)
             self.log.info("{}:{} is DOWN ({}). Connection failed after {} ms".format(addr, port, str(e), length))
-            return Status.DOWN, "{}. Connection failed after {} ms".format(str(e), length)
+            status_checks.append((
+                self.SERVICE_CHECK_CAN_CONNECT,
+                Status.DOWN,
+                "{}. Connection failed after {} ms".format(str(e), length)))
+        else:
+            self.log.debug("{}:{} is UP".format(addr, port))
+            status_checks.append((self.SERVICE_CHECK_CAN_CONNECT, Status.UP, ""))
 
         if response_time:
             self.gauge('network.tcp.response_time', time.time() - start,
                        tags=['url:{}:{}'.format(instance.get('host', None), port),
                              'instance:{}'.format(instance.get('name'))] + custom_tags)
 
-        self.log.debug("{}:{} is UP".format(addr, port))
-        return Status.UP, "UP"
+        if check_certificate_expiration:
+            ssl_check_status, days_left, seconds_left, message = self.check_cert_expiration(
+                url, addr, port, timeout, ca_certs,
+                check_hostname, ssl_server_name,
+                days_warning, days_critical, seconds_warning, seconds_critical,
+                client_key, client_cert)
+            tags_list = custom_tags + [
+                'target_host:{}'.format(url),
+                'port:{}'.format(port),
+                'instance:{}'.format(self.normalize(instance['name']))
+            ]
+            self.gauge('tcp.ssl.days_left', days_left, tags=tags_list)
```

Comment:
very glad you're sending this as a metric, too! was going to ask for this if you hadn't